lib/Wikia/src/Service/User/Permissions/PermissionsServiceImpl.php

```diff
@@ -0,0 +1,391 @@
+<?php
+
+namespace Wikia\Service\User\Permissions;
+
+class PermissionsServiceImpl implements PermissionsService {
+
+	/** @var  PermissionsConfiguration */
+	private $permissionsConfiguration;
+
+	/** @var string[string] - key is user id */
+	private $localExplicitUserGroups = [];
+
+	/** @var string[string] - key is the user id */
+	private $globalExplicitUserGroups = [];
+
+	/** @var string[string] - key is the user id */
+	private $implicitUserGroups = [];
+
+	/** @var string[string] - key is user id */
+	private $userPermissions = [];
+
+	public function __construct() {
+		$this->permissionsConfiguration = new PermissionsConfigurationImpl();
+	}
+
+	public function getConfiguration() {
+		return $this->permissionsConfiguration;
+	}
+
+	public function getExplicitGroups(\User $user ) {
```

Comment:
Ok I guess either way is fine (current solution is one line shorter :D ). So do we have a convention that we want to avoid '\' ? If yes I can change that, as I use it quite a bit in other places.